### PR TITLE
[OIS] Update SDK version

### PR DESCRIPTION
**Pull request by ATmobica**
Update TF-M version to 1.7.0.
Update custom TF-M target AN552.
Adapting examples to new TF-M version.
Refactoring the Cmake build system for passing the SDK targets to the CHIP build in a more generic way.
GN build Cmake file cleanup.
Update IoT socket version.

### CODE REVIEW

File: config/openiotsdk/cmake/sdk.cmake [hunk 1]
```diff
@@ -176,46 +200,85 @@ if(TARGET mbedtls-config)
...
+list(APPEND EXTERNAL_TARGETS
+    iotsdk-blockdevice
+    iotsdk-tdbstore
+)
+
+# Add custom storage library
+add_subdirectory(${OPEN_IOT_SDK_CONFIG}/storage storage_build)
+list(APPEND EXTERNAL_TARGETS
+    openiotsdk-storage
+)
```
Comment (pan-) [inline config/openiotsdk/cmake/sdk.cmake:271]: We can keep it as this is now however I don't think it is correct as it forces includes of target in a specific order. I think a better option would be to set link based on the configuration and not the targets included. 

If the target is not available, the user has to fix it.

**Review by pan- — APPROVED** [review]

File: config/openiotsdk/cmake/sdk.cmake [hunk 2]
```diff
@@ -39,83 +41,105 @@ FetchContent_Declare(
...
+        target_compile_definitions(cmsis-rtos-api
+            INTERFACE
+                CONFIG_RUN_FREERTOS_SECURE_ONLY=$<IF:$<BOOL:${TFM_SUPPORT}>,0,1>
+        )
+    elseif(TARGET cmsis-rtx)
+        target_link_libraries(cmsis-rtos-api
+            INTERFACE
+                cmsis-rtx
+        )
+    endif()
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply config/openiotsdk/cmake/sdk.cmake:142]: Yeah, now we only support the freertos, but when we'll extend this we could use some CMake options or flags.

File: examples/platform/openiotsdk/tf-m/targets/an552/tfm_peripherals_def.h [hunk 3]
```diff
@@ -25,6 +25,9 @@ extern "C" {
 #define TFM_TIMER0_IRQ (TIMER0_IRQn)
 #define TFM_TIMER1_IRQ (TIMER1_IRQn)
 
+#define TFM_FPU_S_TEST_IRQ (GPIO0_0_IRQn)
+#define TFM_FPU_NS_TEST_IRQ (GPIO0_1_IRQn)
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply examples/platform/openiotsdk/tf-m/targets/an552/tfm_peripherals_def.h:29]: From what I understand it's for FPU testing. Disabled by default.

File: examples/platform/openiotsdk/tf-m/targets/an552/device/source/an552_ns_init.c [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+int32_t tfm_ns_platform_init(void)
+{
+    /* Register FPU non-secure test interrupt handler */
+    NVIC_SetVector(TFM_FPU_NS_TEST_IRQ, (uint32_t) TFM_FPU_NS_TEST_Handler);
+
+    /* Enable FPU non-secure test interrupt */
+    NVIC_EnableIRQ(TFM_FPU_NS_TEST_IRQ);
+
+    stdio_init();
```
[the remark this answers is not in the document]
Comment (ATmobica) [reply examples/platform/openiotsdk/tf-m/targets/an552/device/source/an552_ns_init.c:22]: Oh, right. However, it's not a problem in our case when we don't use this function (`tfm_ns_platform_init`). It's used in a default TF-M non-secure app.
Do you mean a ticket in TF-M repo?

**Review by paul-szczepanek-arm — APPROVED** [review]

**Review by pan- — APPROVED** [review]

**Pull request merged**